war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -694,11 +694,17 @@ var jenkinsRules = {
     "INPUT.required" : function(e) { registerRegexpValidator(e,/./,"Field is required"); },
 
 // validate form values to be an integer
-    "INPUT.number" : function(e) { registerRegexpValidator(e,/^\-?(\d+)$/,"Not an integer"); },
-    "INPUT.non-negative-number" : function(e) {
+    "INPUT.number" : function(e) { registerRegexpValidator(e,/^(\d+|)$/,"Not an integer"); },
+    "INPUT.number-required" : function(e) { registerRegexpValidator(e,/^\-?(\d+)$/,"Not an integer"); },
+
+    "INPUT.non-negative-number-required" : function(e) {
         registerRegexpValidator(e,/^\d+$/,"Not a non-negative number");
     },
+
     "INPUT.positive-number" : function(e) {
+        registerRegexpValidator(e,/^(\d*[1-9]\d*|)$/,"Not a positive integer");
```

Comment:
Allowed to begin with leading zeroes why?

Reply:
I assumed it was because a leading 0 with at least one non-zero digit is a valid positive integer.

Reply:
I just restored the original behavior (see #3141).

Reply:
Right, but specifically `\d*` doesn't seem right. Accepting empty values seems fine to restore compatibility, but this? Unclear to me, why.

Reply:
I don't think it's critical. If you think it is necessary, I'll delete `\d*`.

Reply:
@daniel-beck the pattern as a whole does not accept empty values.  It requires at least one digit from the set 1-9.  

I am fine with we disallow a leading zero (thus allowing deletion of the initial `\d*`).  The regular expression will need to have some portion which accepts zero or more digits, but if you feel that leading zeros are not needed, I think it is fine to remove the first `\d*`.

Reply:
> the pattern as a whole does not accept empty values.

Not true. That's the whole reason there's parens around the digits.

Reply:
Ah, you're correct.  Sorry about my poor parsing of the regular expression.

I think an empty field should be allowed so that a default value is assumed when none is provided.

I am indifferent whether leading zeros should be accepted.